crates/cranelift/src/compiler.rs

```diff
@@ -166,19 +166,86 @@ impl wasmtime_environ::Compiler for Compiler {
         self.save_translator(func_translator);
 
         let mut code_buf: Vec<u8> = Vec::new();
-        let mut reloc_sink = RelocSink::new();
-        let mut trap_sink = TrapSink::new();
-        let mut stack_map_sink = StackMapSink::default();
         context
-            .compile_and_emit(
-                isa,
-                &mut code_buf,
-                &mut reloc_sink,
-                &mut trap_sink,
-                &mut stack_map_sink,
-            )
+            .compile_and_emit(isa, &mut code_buf)
             .map_err(|error| CompileError::Codegen(pretty_error(&context.func, error)))?;
 
+        let result = context.mach_compile_result.as_ref().unwrap();
+
+        let mut func_relocs = Vec::new();
+        for &MachReloc {
+            offset,
+            srcloc: _,
+            kind,
+            ref name,
+            addend,
+        } in result.buffer.relocs()
+        {
```

Comment:
Can we factor this loop out into a helper of some sort? I think it helped readability in the original code to have things factored as a consequence of separate trait impls; it might be nice to have e.g. `mach_reloc_to_reloc()` and map the vector through it or something similar.

Reply:
Done